PyPDF2 doesn't install pillow

Running PyPDF2 from scratch and trying to extract images (via `page.images` property) throws ModuleNotFoundError.

<img width="754" alt="image" src="https://user-images.githubusercontent.com/2257119/195187341-da6de5bb-e52e-441c-8cb3-c111abb964ee.png">

This is because my current setup doesn't have pillow installed, _and PyPDF2 doesn't install it_. My proposal is to either add an extra to the setuptools config that installs pillow, or to have this requirement more visible in the docs. 

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
macOS-12.6-x86_64-i386-64bit

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.11.1
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
with open("2210.03747.pdf", "rb") as your_pdf_file:
    pdf = PdfReader(your_pdf_file)
    for page in pdf.pages:
        for image in page.images:
            print(image)
```

Any PDF with xObject images; for example: https://arxiv.org/pdf/2210.03747.pdf

## Traceback

This is the complete Traceback I see:

```
x_object_obj = {'/Type': '/XObject', '/Subtype': '/Image', '/Width': 417, '/Height': 328, '/BitsPerComponent': 8, '/ColorSpace': '/DeviceRGB', '/SMask': IndirectObject(33, 0, 4575023296), '/Filter': '/FlateDecode'}

    def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes]:
        """
        Users need to have the pillow package installed.
    
        It's unclear if PyPDF2 will keep this function here, hence it's private.
        It might get removed at any point.
    
        :return: Tuple[file extension, bytes]
        """
>       from PIL import Image
E       ModuleNotFoundError: No module named 'PIL'

venv/lib/python3.10/site-packages/PyPDF2/filters.py:567: ModuleNotFoundError
```

--- a/PyPDF2/_page.py
+++ b/PyPDF2/_page.py
@@ -373,6 +373,8 @@ class PageObject(DictionaryObject):
         """
         Get a list of all images of the page.
 
+        This requires pillow. You can install it via 'pip install PyPDF2[image]'.
+
         For the moment, this does NOT include inline images. They will be added
         in future.
         """

--- a/PyPDF2/filters.py
+++ b/PyPDF2/filters.py
@@ -564,7 +564,13 @@ def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes]:
 
     :return: Tuple[file extension, bytes]
     """
-    from PIL import Image
+    try:
+        from PIL import Image
+    except ImportError:
+        raise ImportError(
+            "pillow is required to do image extraction. "
+            "It can be installed via 'pip install PyPDF2[image]'"
+        )
 
     size = (x_object_obj[IA.WIDTH], x_object_obj[IA.HEIGHT])
     data = x_object_obj.get_data()  # type: ignore

--- a/docs/user/encryption-decryption.md
+++ b/docs/user/encryption-decryption.md
@@ -1,7 +1,7 @@
 # Encryption and Decryption of PDFs
 
 Please see the note in the
-[Installation doc](https://pypdf2.readthedocs.io/en/latest/user/installation.html)
+[Installation doc](installation.html)
 for installing the extra dependencies if interacting with PDFs that use AES.
 
 ## Encrypt

--- a/docs/user/encryption-decryption.md
+++ b/docs/user/encryption-decryption.md
@@ -1,8 +1,7 @@
 # Encryption and Decryption of PDFs
 
-Please see the note in the
-[Installation doc](installation.html)
-for installing the extra dependencies if interacting with PDFs that use AES.
+> Please see the note in the [installation guide](installation.html)
+> for installing the extra dependencies if interacting with PDFs that use AES.
 
 ## Encrypt
 

--- a/docs/user/extract-images.md
+++ b/docs/user/extract-images.md
@@ -1,5 +1,8 @@
 # Extract Images
 
+> Please note: In order to use the following code you need to install optional
+> dependencies, see [installation guide](installation.html).
+
 Every page of a PDF document can contain an arbitrary amount of images.
 The names of the files may not be unique.
 

--- a/docs/user/installation.md
+++ b/docs/user/installation.md
@@ -20,6 +20,20 @@ install PyPDF2 for your current user:
 pip install --user PyPDF2
 ```
 
+### Optional dependencies
+
+PyPDF2 tries to be as self-contained as possible, but for some tasks the amout
+of work to properly maintain the code would be too high. This is especially the
+case for cryptography and image formats.
+
+If you simply want to unstall all optional dependencies, please run
+
+```
+pip install PyPDF2[full]
+```
+
+Alternatively, you can install just some:
+
 If you plan to use PyPDF2 for encrypting or decrypting PDFs that use AES, you
 will need to install some extra dependencies. Encryption using RC4 is supported
 using the regular installation.
@@ -28,6 +42,11 @@ using the regular installation.
 pip install PyPDF2[crypto]
 ```
 
+If you plan to use image extraction, you need Pillow:
+
+```
+pip install PyPDF2[image]
+```
 
 ## Python Version Support
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -42,9 +42,7 @@ install_requires =
     dataclasses; python_version < '3.7'
 
 [options.extras_require]
-full =
-    PyCryptodome
-    Pillow
+full = PyCryptodome; Pillow
 crypto = PyCryptodome
 image = Pillow
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -42,7 +42,11 @@ install_requires =
     dataclasses; python_version < '3.7'
 
 [options.extras_require]
+full =
+    PyCryptodome
+    Pillow
 crypto = PyCryptodome
+image = Pillow
 
 [mutmut]
 backup = False

--- a/tests/test_filters.py
+++ b/tests/test_filters.py
@@ -1,4 +1,5 @@
 import string
+import sys
 from io import BytesIO
 from itertools import product as cartesian_product
 from unittest.mock import patch
@@ -226,3 +227,20 @@ def test_issue_399():
     name = "tika-976970.pdf"
     reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
     reader.pages[1].extract_text()
+
+
+def test_image_without_imagemagic():
+    with patch.dict(sys.modules):
+        sys.modules["PIL"] = None
+        url = "https://corpora.tika.apache.org/base/docs/govdocs1/914/914102.pdf"
+        name = "tika-914102.pdf"
+        data = BytesIO(get_pdf_from_url(url, name=name))
+        reader = PdfReader(data, strict=True)
+
+        for page in reader.pages:
+            with pytest.raises(ImportError) as exc:
+                page.images
+            assert (
+                exc.value.args[0]
+                == "pillow is required to do image extraction. It can be installed via 'pip install PyPDF2[image]'"
+            )

--- a/tests/test_reader.py
+++ b/tests/test_reader.py
@@ -5,7 +5,6 @@ from io import BytesIO
 from pathlib import Path
 
 import pytest
-from PIL import Image
 
 from PyPDF2 import PdfReader
 from PyPDF2._reader import convert_to_int, convertToInt
@@ -188,6 +187,8 @@ def test_get_outline(src, outline_elements):
     ],
 )
 def test_get_images(src, expected_images):
+    from PIL import Image
+
     src_abs = RESOURCE_ROOT / src
     reader = PdfReader(src_abs)
 

